src/main/java/org/folio/helper/CheckinHelper.java

```diff
@@ -217,13 +218,23 @@ protected Future<Boolean> receiveInventoryItemAndUpdatePiece(JsonObject item, Pi
       })
       // Add processing error if item failed to be updated
       .onFailure(e -> {
-        logger.error("Item associated with piece '{}' cannot be updated", piece.getId());
-        addError(piece.getPoLineId(), piece.getId(), ITEM_UPDATE_FAILED.toError());
+        produceErrorMessage(piece, e.toString());
         promise.complete(false);
       });
     return promise.future();
   }
 
+  // This function produces error message based on the error it receives from the checkinItem()
```

Comment:
propose to rename function to be more self descriptive instead of adding comment  here.
addErrorForUpdatingItem()

Reply:
yep, will do